### PR TITLE
Feat/add json logger

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Add a json logger to marl-eval to enable easy logging according to the protocol in the correct json format.

### CODE REVIEW

**Review by sash-a — CHANGES_REQUESTED** [review]
Looking great Ruan pretty much entirely just docstring changes. 

The big thing to clarify is `evaluation_step`, we want to make it clear that it's not the number of evaluation timesteps.

File: marl_eval/json_tools/json_logger.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                "elapsed_time": current_time - self.start_time,
+            } | metrics
+            step_str = f"step_{evaluation_step}"
+            if step_str in self.run_data:
+                self.run_data[step_str].update(step_metrics)
+            else:
+                self.run_data[step_str] = step_metrics
+
+        with open(self.file_path, "w") as f:
+            json.dump(self.data, f, indent=4)
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply marl_eval/json_tools/json_logger.py:115]: True. We've just used 4 as a default in the past.

File: marl_eval/json_tools/json_logger.py [hunk 2]
```diff
@@ -19,19 +19,19 @@
...
+    from the implementation found in BenchMARL which can be viewed at:
+    https://tinyurl.com/2t6fy548
 
     Args:
-        path (str): where to write the file
-        algorithm_name (str): algorithm name
-        task_name (str): task name
-        environment_name (str): environment name
-        seed (int): random seed of the experiment
+        path (str): folder path for where to save metrics.json.
```
Comment (sash-a) [inline marl_eval/json_tools/json_logger.py:30]: Sorry my mistake, please change it again :smile: 

```suggestion
        path (str): folder path for saving the `metrics.json` file.
```

**Review by sash-a — APPROVED** [review]
Tiny thing, but approving anyways, feel free to make this change and then merge (changing indent level is up to you) :smile:

**Pull request merged**